### ISSUE TITLE
Output to CLI is thirce

### ISSUE THREAD

**Issue by rdpanek** [webdriverio/webdriverio#3298]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.2.8
 - **Mode:** [Testrunner]
 - **If WDIO Testrunner, running sync/async:** sync: true
 - **Node.js version:** 10.9.0
 - **NPM version:** 6.2.0
 - **Browser name and version:** Chrome 71
 - **Platform name and version:** OSX Mojave 10.14.1 
 - **Additional wdio packages used (if applicable):
```
  "devDependencies": {
    "@babel/cli": "7.2.3",
    "@babel/core": "7.2.2",
    "@babel/preset-env": "7.2.3",
    "@babel/register": "7.0.0",
    "@wdio/cli": "5.0.3",
    "@wdio/local-runner": "5.1.0",
    "@wdio/mocha-framework": "5.1.0",
    "@wdio/sync": "5.1.0",
    "@wdio/spec-reporter": "5.0.3",
    "aws-sdk": "2.0.9",
    "chai": "4.1.2",
    "node-uuid": "1.4.8",
    "webdriverio": "5.1.0"
  }
```

**Config of WebdriverIO**
wdio.conf.js
```
reporters: ['spec'],
```

**Describe the bug**
 test is run once but output to CLI is thirce ( reporter / log ) why?

```
...
 RUNNING  0-0 in chrome - /test/my/smoke.js

Test Suites:     0 passed, 1 total (0% completed)
Time:         🕔  11.57s
...

"spec" Reporter:
------------------------------------------------------------------
[chrome 71.0.3578.98 Windows NT #0-0] Spec:
...
Stdout:
2019-01-10T11:48:40.646Z INFO wdio-cli:Launcher:  Run onPrepare hook
...
"spec" Reporter:
------------------------------------------------------------------
[chrome 71.0.3578.98 Windows NT #0-0] Spec:
...
Stdout:
2019-01-10T11:48:40.646Z INFO wdio-cli:Launcher:  Run onPrepare hook
...
"spec" Reporter:
------------------------------------------------------------------
[chrome 71.0.3578.98 Windows NT #0-0] Spec:
...
Stdout:
2019-01-10T11:53:26.608Z INFO wdio-cli:Launcher:  Run onPrepare hook
...
2019-01-10T11:53:39.105Z DEBUG wdio-local-runner: Runner 0-0 finished with exit code 1

Test Suites:     0 passed, 1 failed, 1 total (100% completed)
Time:         🕔  12.65s
```

**Expected behavior**
Exist some solutions how to fix? because output is a big. I expected only once, Thx

**Log**

**Additional context**
How's a right syntax, or why my hooks not called?

**Comment by abjerstedt** [comment]
This is a known issue,  l and needs to be resolved.

Side note: please go through and update your wdio packages to the most current version. Since the launch of v5 we've been moving quickly with fixes.

**Comment by mgrybyk** [comment]
same here https://github.com/webdriverio/webdriverio/issues/3188

**Comment by mgrybyk** [comment]
I think this one is pretty difficult one and might require logging refactoring

**Comment by abjerstedt** [comment]
@mgrybyk Christian believes its related to the interface not properly clearing.  I spent a decent amount of time looking into this last week, and couldn't isolate a quick fix.

**Comment by mgrybyk** [comment]
@abjerstedt I wish there were an ability to turn off this clearing feature and use regular old school append logger :)

Multiple results is only a part of problem. Another one is that it is not showing test output while test is running, only in the end.

Also, you can't clear console output for CI system like Jenkins, so every updateView() will result output duplication.

**Comment by abjerstedt** [comment]
@mgrybyk Yep. All 3 things you listed are known issues that we want to eventually address.

**Comment by rdpanek** [comment]
@mgrybyk Yep, I agree with you. "Old school" logger it's better.

**Comment by abjerstedt** [comment]
I'm still working on this. I need to track down whats actually causing it to print 3x.

**Comment by mgrybyk** [comment]
Something in updateView method :)

**Comment by abjerstedt** [comment]
As a quick and dirty, I think we can just clear the buffers each time we print (it seems to be working). However, spec reporter has a symptom i need to fix before submitting this PR.

**Comment by mgrybyk** [comment]
I was trying to remove redundant calls, but in some cases results were still duplicated.

Please try case were you have at least two tests and at least one failed and one passed tests

**Comment by abjerstedt** [comment]
@mgrybyk Yep. thats what blocked me last week. The fix is honestly to clear the buffer after every print. This solution fixes it for non TTY environments too. 

The problem is that i'm losing part of the initial report print atm (probably related to the clear).

**Comment by abjerstedt** [comment]
@mgrybyk @LiXii I have a proposed fix linked above. This won't make console log in RT, but it will fix the triple printing, and should also help in CI/CD environments.

**Comment by TC-rGangal** [comment]
@abjerstedt thanks for merging the PR https://github.com/webdriverio/webdriverio/pull/3377. I tried getting the latest build this morning of webdriverio. But the issue still exists. 
I just did ```npm i webdriverio ```

**Comment by abjerstedt** [comment]
@TC-rGangal you need to remember to update all the packages. The fix was in @wdio/cli

**Comment by TC-rGangal** [comment]
my bad.. thanks @abjerstedt

**Comment by rdpanek** [comment]
@abjerstedt Hi, I updated all dependencies, but problem still exist:
Dependencies
```
  "@babel/cli": "7.2.3",
    "@babel/core": "7.2.2",
    "@babel/preset-env": "7.2.3",
    "@babel/register": "7.0.0",
    "@wdio/cli": "5.6.4",
    "@wdio/local-runner": "5.6.4",
    "@wdio/mocha-framework": "5.1.0",
    "@wdio/spec-reporter": "5.6.4",
    "@wdio/sync": "5.6.4",
    "aws-sdk": "2.0.9",
    "chai": "4.1.2",
    "node-uuid": "1.4.8",
    "webdriverio": "5.6.4"
```
and output is still duplicated - what next?

**Comment by abjerstedt** [comment]
@rdpanek My PR resolved the issue. There have been subsequent PRs that introduced something similar back. (You notice that reporters still only log once).

SideNote: Try to keep all packages in sync that you can (like mocha-framework)

**Comment by rdpanek** [comment]
Hi @abjerstedt I still working on investigate this bug and in default terminal on OSX is working properly, but in iterm2 is stil output duplicated.
